editorconfig.go

```diff
@@ -193,71 +193,79 @@ func (d *Definition) merge(md *Definition) {
 	}
 }
 
-// InsertToIniFile ... TODO
-func (d *Definition) InsertToIniFile(iniFile *ini.File) {
-	iniSec := iniFile.Section(d.Selector)
-	for k, v := range d.Raw {
-		if k == "insert_final_newline" {
-			if d.InsertFinalNewline != nil {
-				iniSec.Key(k).SetValue(strconv.FormatBool(*d.InsertFinalNewline))
-			} else {
-				insertFinalNewline, ok := d.Raw["insert_final_newline"]
-				if ok {
-					iniSec.Key(k).SetValue(strings.ToLower(insertFinalNewline))
-				}
-			}
-		} else if k == "trim_trailing_whitespace" {
-			if d.TrimTrailingWhitespace != nil {
-				iniSec.Key(k).SetValue(strconv.FormatBool(*d.TrimTrailingWhitespace))
-			} else {
-				trimTrailingWhitespace, ok := d.Raw["trim_trailing_whitespace"]
-				if ok {
-					iniSec.Key(k).SetValue(strings.ToLower(trimTrailingWhitespace))
-				}
+func setValues(d *Definition, iniSection *ini.Section, key string, value string) {
+	if key == "insert_final_newline" {
+		if d.InsertFinalNewline != nil {
+			iniSection.Key(key).SetValue(strconv.FormatBool(*d.InsertFinalNewline))
+		} else {
+			insertFinalNewline, ok := d.Raw["insert_final_newline"]
+			if ok {
+				iniSection.Key(key).SetValue(strings.ToLower(insertFinalNewline))
 			}
-		} else if k == "charset" {
-			iniSec.Key(k).SetValue(d.Charset)
-		} else if k == "end_of_line" {
-			iniSec.Key(k).SetValue(d.EndOfLine)
-		} else if k == "indent_style" {
-			iniSec.Key(k).SetValue(d.IndentStyle)
-		} else if k == "tab_width" {
-			tabWidth, ok := d.Raw["tab_width"]
-			if ok && tabWidth == "unset" {
-				iniSec.Key(k).SetValue(tabWidth)
-			} else {
-				iniSec.Key(k).SetValue(strconv.Itoa(d.TabWidth))
+		}
+	} else if key == "trim_trailing_whitespace" {
+		if d.TrimTrailingWhitespace != nil {
+			iniSection.Key(key).SetValue(strconv.FormatBool(*d.TrimTrailingWhitespace))
+		} else {
+			trimTrailingWhitespace, ok := d.Raw["trim_trailing_whitespace"]
+			if ok {
+				iniSection.Key(key).SetValue(strings.ToLower(trimTrailingWhitespace))
 			}
-		} else if k == "indent_size" {
-			iniSec.Key(k).SetValue(d.IndentSize)
+		}
+	} else if key == "charset" {
+		iniSection.Key(key).SetValue(d.Charset)
+	} else if key == "end_of_line" {
+		iniSection.Key(key).SetValue(d.EndOfLine)
+	} else if key == "indent_style" {
+		iniSection.Key(key).SetValue(d.IndentStyle)
+	} else if key == "tab_width" {
+		tabWidth, ok := d.Raw["tab_width"]
+		if ok && tabWidth == "unset" {
+			iniSection.Key(key).SetValue(tabWidth)
 		} else {
-			iniSec.Key(k).SetValue(v)
+			iniSection.Key(key).SetValue(strconv.Itoa(d.TabWidth))
 		}
+	} else if key == "indent_size" {
+		iniSection.Key(key).SetValue(d.IndentSize)
+	} else {
+		iniSection.Key(key).SetValue(value)
 	}
```

Comment:
could it be a big switch statement?

Reply:
Nope, as any `case`-statement also adds one to the complexity calculation. In fact, when porting this piece of code to a `switch`-statement it gets one point worse in complexity. I guess because the `switch` and `case` are counted separately.

Reply:
`gocritic` would probably disagree with the non-usage of `switch`/`case` here.

https://github.com/go-critic/go-critic

Reply:
Okay, I could change that but I see you've already did it in #26

Reply:
Some of it only. The split of functions you've done is great though.

Reply:
So - what should I do with that PR now? 
Branch from your PR and do a PR against your branch for the split?